enterprise/frontend/src/embedding-sdk/components/public/dashboard/SdkDashboard.tsx

```diff
@@ -232,7 +241,29 @@ const SdkDashboardInner = ({
           : onNavigateToNewCardFromDashboard
       }
       onNewQuestion={() => {
-        setRenderMode("queryBuilder");
+        if (isDashboardDirty) {
+          show({
```

Comment:
Use the exact same copy as the core dashboard.
https://github.com/metabase/metabase/blob/afded810c1b46b53b1ebf2c3c77875f0b3c6cc44/frontend/src/metabase/dashboard/components/DashboardLeaveConfirmationModal/DashboardLeaveConfirmationModal.tsx#L37-L43